_posts/mattox-endpoints.md

```diff
@@ -0,0 +1,184 @@
+
+## Further customization
+
+Not everything can be configured through string values, though. If that's the case, Mattox allows accessing the usual NServiceBus code-based configuration in a couple of ways.
+
+For specific customizations, the endpoint API exposes several options. For example, the following snippet allows replacing the default serializer with the JSON.Net one:
+
+```c#
+var endpoint = new AmazonSqsEndpoint("my-endpoint");
+endpoint.ReplaceDefaultSerializer<NewtonsoftJsonSerializer>();
+var endpointInstance = await endpoint.Start();
+```
+
+Or we can customize the transport by using the following code:
+
+```c#
+var endpoint = new AmazonSqsEndpoint("my-endpoint");
+endpoint.CustomizeTransport(transport =>
+{
+   // apply transport settings
+});
+var endpointInstance = await endpoint.Start();
+```
+
+The transport customization delegate is invoked after applying any `IConfiguration`-defined transport setting.
+
+If the need is to apply changes to the whole endpoint configuration before starting it, we can preview the `EndpointConfiguration` instance before it's used:
+
+```c#
+var endpoint = new AmazonSqsEndpoint("my-endpoint");
+endpoint.PreviewConfiguration(endpointConfiguration =>
+{
+   // Apply the needed changes
+});
+var endpointInstance = await endpoint.Start();
+```
+
+As for the transport customization delegate, the endpoint configuration one is invoked after applying any `IConfiguration`-defined setting.
+
+At the moment, Mattox endpoints are primarily experiments, so I've only implemented support for RabbitMQ and AmazonSQS, which are different enough to validate the approach.
+
+> Let me know in the comments below if that's something you might be interested in using, and I'll bring it to completion and publish it to NuGet.
+
+## That's _not_ all, folks!
+
+What if it would be possible to do something like the following:
+
+```dockerfile
+FROM NServiceBus.Endpoints:9 AS base
+
+# use a dedicated image to build and publish
+FROM mcr.microsoft.com/dotnet/sdk:8.0 AS build
+WORKDIR /src
+COPY . .
+RUN dotnet publish -c Release -o /app
+
+FROM base AS final
+WORKDIR /app
+COPY --from=build /app .
+
+ENV NServiceBus__EndpointName=MyEndpointName
+ENV NServiceBus__Transport=AmazonSQS
+
+# Transport specific settings
+ENV AWS_ACCESS_KEY_ID=<access key ID>
+ENV AWS_SECRET_ACCESS_KEY=<secret access key>
+ENV AWS_REGION=<region>
+```
+
+> Let me know in the comments below what you think about this option too.
+
+The above Dockerfile defines a containerized endpoint, where everything comes from the `base` layer, and the user code can be limited to message handlers and sagas.
+
+The endpoint configuration is expressed via environment variables, including transport selection, persistence, serialization, and other NServiceBus options.
+
+[Tomek](https://twitter.com/Masternak) and I built a far-from-being-production-ready spike to demonstrate how it could work.
```

Comment:
Might be nice to link to it here instead of teasing it, if possible

Reply:
That would be great, but it's not in a publicly available repo.